dynamic_programming/cut_rod.cpp

```diff
@@ -70,25 +70,21 @@ int maxProfitByCuttingRod(const std::array<int, T> &price, const int n) {
  */
 static void test() {
     // Test 1
-    const int n1 = 8;                                           // size of rod
-    std::array<int, n1> price1 = {1, 5, 8, 9, 10, 17, 17, 20};  // price array
-    const int max_profit1 =
+    const int16_t n1 = 8;                                        // size of rod
+   std::array<int32_t, n1> price1 = {1,2,4,6,8,45,21,9};  // price array
+   const int64_t max_profit1 =
         dynamic_programming::cut_rod::maxProfitByCuttingRod(price1, n1);
-    const int expected_max_profit1 = 22;
-    assert(max_profit1 == expected_max_profit1);
     std::cout << "Maximum profit with " << n1 << " inch road is " << max_profit1
               << std::endl;
 
     // Test 2
-    const int n2 = 30;  // size of rod
-    std::array<int, n2> price2 = {
+    const int16_t n2 = 30;  // size of rod
+    std::array<int32_t, n2> price2 = {
         1,  5,  8,  9,  10, 17, 17, 20, 24, 30,  // price array
         31, 32, 33, 34, 35, 36, 37, 38, 39, 40,
         41, 42, 43, 44, 45, 46, 47, 48, 49, 50};
-    const int max_profit2 =
+    const int64_t max_profit2 =
         dynamic_programming::cut_rod::maxProfitByCuttingRod(price2, n2);
-    const int expected_max_profit2 = 90;
-    assert(max_profit2 == expected_max_profit2);
```

Comment:
Please re-add this, this are assert checks, they are **required** in every algorithm.